awx_collection/tests/integration/targets/tower_notification/tasks/main.yml

```diff
@@ -9,14 +9,78 @@
     hipchat_not: "AWX-Collection-tests-tower_notification-hipchat-not-{{ lookup('password', '/dev/null chars=ascii_letters length=16') }}"
     irc_not: "AWX-Collection-tests-tower_notification-irc-not-{{ lookup('password', '/dev/null chars=ascii_letters length=16') }}"
 
-- name: Create Slack notification
+- name: Test deprication warnings
+  tower_notification:
+    name: "{{ slack_not }}"
+    organization: Default
+    notification_type: slack
+    username: maw
+    sender: maw
+    recipients:
+      - everyone
+    use_tls: True
+    host: all
+    use_ssl: False
+    password: password
+    port: 12
+    channels:
+      - general
+    token: chunkecheese
+    account_token: asdf1234
+    from_number: "1 (888) 733-4281"
+    to_numbers:
+      - 867-5309
+    account_sid: vicious
+    subdomain: 'redhat.com'
+    service_key: skeleton
+    client_name: Bill
+    message_from: me
+    api_url: https://tower.example.com/api/v2
+    color: green
+    rooms:
+      - The Study
+      - Kitchen
+      - Ballroom
+      - Conservatory
+      - Billiard Room
+      - Library
+      - Hall
+      - Lounge
+      - Dining Room
+      - Cellar
+    notify: True
+    url: ansible.com
+    headers:
+      X-Custom-Header: value123
+    server: littimer.somewhere.com
+    nickname: chalk
+    targets:
+      - zombie
+    state: absent
+  register: result
+  ignore_errors: True
+
+- assert:
+    that:
+      - "'deprecations' in result"
+      # The 27 can be count from the size of the OLD_INPUT_NAMES list in the module
+      - result['deprecations'] | length() == 27
+
+- name: Create Slack notification with custom messages
   tower_notification:
     name: "{{ slack_not }}"
     organization: Default
     notification_type: slack
     token: a_token
     channels:
       - general
+    messages:
+      started:
+        message: "{{ '{{' }} job_friendly_name {{' }}' }} {{ '{{' }} job.id {{' }}' }} started"
```

Comment:
the example playbook I commented with has an alternative for this syntax that doesn't look quite _as_ bad. I would replace this task with that one.

Reply:
Done. Since I was in there I also took care of the headers default and being able to delete w/o credential_type.